### PR TITLE
XW-2567 | VE - update anon warning messages

### PR DESCRIPTION
**Pull request by kvas-damian**
https://wikia-inc.atlassian.net/browse/XW-2567

* more user friendly messages informing user that as anon not all features are enabled
* separate login and register button

Ping @Wikia/x-wing

### CODE REVIEW

**Comment by WikiaTech** [comment]
SonarQube analysis reported 11 issues
* MAJOR 8 major
* MINOR 2 minor
* INFO 1 info

#### Top 10 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. MAJOR [ve.ui.WikiaMediaInsertDialog.js#L14](https://github.com/Wikia/app/blob/6215798e847e55c55101a63899842e6f61b3f96e/extensions/VisualEditor/wikia/modules/ve/ui/dialogs/ve.ui.WikiaMediaInsertDialog.js#L14): Make this function anonymous by removing its name: 'function() {...}'. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3ANamedFunctionExpression)
1. MAJOR [ve.ui.WikiaMediaInsertDialog.js#L273](https://github.com/Wikia/app/blob/6215798e847e55c55101a63899842e6f61b3f96e/extensions/VisualEditor/wikia/modules/ve/ui/dialogs/ve.ui.WikiaMediaInsertDialog.js#L273): Remove this commented out code. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3ACommentedCode)
1. MAJOR [ve.ui.WikiaMediaInsertDialog.js#L625](https://github.com/Wikia/app/blob/6215798e847e55c55101a63899842e6f61b3f96e/extensions/VisualEditor/wikia/modules/ve/ui/dialogs/ve.ui.WikiaMediaInsertDialog.js#L625): Restrict what this loop acts on by testing each property. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AForIn)
1. MAJOR [ve.ui.WikiaMediaInsertDialog.js#L650](https://github.com/Wikia/app/blob/6215798e847e55c55101a63899842e6f61b3f96e/extensions/VisualEditor/wikia/modules/ve/ui/dialogs/ve.ui.WikiaMediaInsertDialog.js#L650): Restrict what this loop acts on by testing each property. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AForIn)
1. MAJOR [ve.ui.WikiaMediaInsertDialog.js#L653](https://github.com/Wikia/app/blob/6215798e847e55c55101a63899842e6f61b3f96e/extensions/VisualEditor/wikia/modules/ve/ui/dialogs/ve.ui.WikiaMediaInsertDialog.js#L653): Remove this assignment of "type". [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AS2310)
1. MAJOR [ve.ui.WikiaVideoInsertDialog.js#L14](https://github.com/Wikia/app/blob/6215798e847e55c55101a63899842e6f61b3f96e/extensions/VisualEditor/wikia/modules/ve/ui/dialogs/ve.ui.WikiaVideoInsertDialog.js#L14): Make this function anonymous by removing its name: 'function() {...}'. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3ANamedFunctionExpression)
1. MAJOR [ve.ui.WikiaAnonWarningWidget.js#L1](https://github.com/Wikia/app/blob/6215798e847e55c55101a63899842e6f61b3f96e/extensions/VisualEditor/wikia/modules/ve/ui/widgets/ve.ui.WikiaAnonWarningWidget.js#L1): Make this function anonymous by removing its name: 'function() {...}'. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3ANamedFunctionExpression)
1. MAJOR [ve.ui.WikiaUploadWidget.js#L14](https://github.com/Wikia/app/blob/6215798e847e55c55101a63899842e6f61b3f96e/extensions/VisualEditor/wikia/modules/ve/ui/widgets/ve.ui.WikiaUploadWidget.js#L14): Make this function anonymous by removing its name: 'function() {...}'. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3ANamedFunctionExpression)
1. MINOR [VisualEditor.php#L45](https://github.com/Wikia/app/blob/6215798e847e55c55101a63899842e6f61b3f96e/extensions/VisualEditor/wikia/VisualEditor.php#L45): Define a constant instead of duplicating this literal "scripts" 3 times. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1192)
1. MINOR [VisualEditor.php#L46](https://github.com/Wikia/app/blob/6215798e847e55c55101a63899842e6f61b3f96e/extensions/VisualEditor/wikia/VisualEditor.php#L46): Define a constant instead of duplicating this literal "dependencies" 3 times. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1192)

**Comment by WikiaTech** [comment]
SonarQube analysis reported 11 issues
* MAJOR 8 major
* MINOR 2 minor
* INFO 1 info

#### Top 10 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. MAJOR [ve.ui.WikiaMediaInsertDialog.js#L14](https://github.com/Wikia/app/blob/76961182d7f888b22d8e4432f9647fbf149ca2a2/extensions/VisualEditor/wikia/modules/ve/ui/dialogs/ve.ui.WikiaMediaInsertDialog.js#L14): Make this function anonymous by removing its name: 'function() {...}'. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3ANamedFunctionExpression)
1. MAJOR [ve.ui.WikiaMediaInsertDialog.js#L273](https://github.com/Wikia/app/blob/76961182d7f888b22d8e4432f9647fbf149ca2a2/extensions/VisualEditor/wikia/modules/ve/ui/dialogs/ve.ui.WikiaMediaInsertDialog.js#L273): Remove this commented out code. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3ACommentedCode)
1. MAJOR [ve.ui.WikiaMediaInsertDialog.js#L625](https://github.com/Wikia/app/blob/76961182d7f888b22d8e4432f9647fbf149ca2a2/extensions/VisualEditor/wikia/modules/ve/ui/dialogs/ve.ui.WikiaMediaInsertDialog.js#L625): Restrict what this loop acts on by testing each property. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AForIn)
1. MAJOR [ve.ui.WikiaMediaInsertDialog.js#L650](https://github.com/Wikia/app/blob/76961182d7f888b22d8e4432f9647fbf149ca2a2/extensions/VisualEditor/wikia/modules/ve/ui/dialogs/ve.ui.WikiaMediaInsertDialog.js#L650): Restrict what this loop acts on by testing each property. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AForIn)
1. MAJOR [ve.ui.WikiaMediaInsertDialog.js#L653](https://github.com/Wikia/app/blob/76961182d7f888b22d8e4432f9647fbf149ca2a2/extensions/VisualEditor/wikia/modules/ve/ui/dialogs/ve.ui.WikiaMediaInsertDialog.js#L653): Remove this assignment of "type". [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AS2310)
1. MAJOR [ve.ui.WikiaVideoInsertDialog.js#L14](https://github.com/Wikia/app/blob/76961182d7f888b22d8e4432f9647fbf149ca2a2/extensions/VisualEditor/wikia/modules/ve/ui/dialogs/ve.ui.WikiaVideoInsertDialog.js#L14): Make this function anonymous by removing its name: 'function() {...}'. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3ANamedFunctionExpression)
1. MAJOR [ve.ui.WikiaAnonWarningWidget.js#L1](https://github.com/Wikia/app/blob/76961182d7f888b22d8e4432f9647fbf149ca2a2/extensions/VisualEditor/wikia/modules/ve/ui/widgets/ve.ui.WikiaAnonWarningWidget.js#L1): Make this function anonymous by removing its name: 'function() {...}'. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3ANamedFunctionExpression)
1. MAJOR [ve.ui.WikiaUploadWidget.js#L14](https://github.com/Wikia/app/blob/76961182d7f888b22d8e4432f9647fbf149ca2a2/extensions/VisualEditor/wikia/modules/ve/ui/widgets/ve.ui.WikiaUploadWidget.js#L14): Make this function anonymous by removing its name: 'function() {...}'. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3ANamedFunctionExpression)
1. MINOR [VisualEditor.php#L45](https://github.com/Wikia/app/blob/76961182d7f888b22d8e4432f9647fbf149ca2a2/extensions/VisualEditor/wikia/VisualEditor.php#L45): Define a constant instead of duplicating this literal "scripts" 3 times. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1192)
1. MINOR [VisualEditor.php#L46](https://github.com/Wikia/app/blob/76961182d7f888b22d8e4432f9647fbf149ca2a2/extensions/VisualEditor/wikia/VisualEditor.php#L46): Define a constant instead of duplicating this literal "dependencies" 3 times. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1192)

File: extensions/VisualEditor/wikia/modules/ve/ui/styles/ve.ui.WikiaMediaInsertDialog.scss [hunk 1]
```diff
@@ -370,3 +370,7 @@ $width-cart: 120px;
 		width: 32px;
 	}
 }
+
+.ve-ui-wikiaRegisterButton {
```
Comment (hakubo) [inline extensions/VisualEditor/wikia/modules/ve/ui/styles/ve.ui.WikiaMediaInsertDialog.scss:374]: camel case or dashes?

Comment (kvas-damian) [reply extensions/VisualEditor/wikia/modules/ve/ui/styles/ve.ui.WikiaMediaInsertDialog.scss:374]: It's hard to say. I quickly scanned this file and didn't find any convention...

File: extensions/VisualEditor/wikia/modules/ve/ui/widgets/ve.ui.WikiaAnonWarningWidget.js [hunk 2]
```diff
@@ -11,21 +11,38 @@ ve.ui.WikiaAnonWarningWidget.prototype.setupAnonWarning = function (predecessor
...
+			'wikia-visualeditor-anon-warning',
+			loginLink.get(0).outerHTML,
+			registerLink.get(0).outerHTML
+		) );
+
+	this.$anonWarning.find('.ve-ui-wikia-anon-warning__login-link')
+		.on( 'click', function () {
+			window.wikiaAuthModal.load( {
+				url: '/signin',
+				onAuthSuccess: function () {
```
Comment (hakubo) [inline extensions/VisualEditor/wikia/modules/ve/ui/widgets/ve.ui.WikiaAnonWarningWidget.js]: I think we can ommit the function here by:
```js
this.emit.bind(this, 'logInSuccess')
```

File: extensions/VisualEditor/wikia/modules/ve/ui/widgets/ve.ui.WikiaUploadWidget.js [hunk 3]
```diff
@@ -243,13 +250,16 @@ ve.ui.WikiaUploadWidget.prototype.setupForLoggedOut = function () {
 		this.emit( 'logInClick' );
 	}.bind( this ) );
 
+	this.registerButton.on( 'click', function () {
+		this.emit( 'registerClick' );
```
Comment (hakubo) [inline extensions/VisualEditor/wikia/modules/ve/ui/widgets/ve.ui.WikiaUploadWidget.js]: this event name might be misleading

Comment (hakubo) [reply extensions/VisualEditor/wikia/modules/ve/ui/widgets/ve.ui.WikiaUploadWidget.js]: registerButtonClicked maybe?

**Comment by WikiaTech** [comment]
SonarQube analysis reported 11 issues
* MAJOR 8 major
* MINOR 2 minor
* INFO 1 info

#### Top 10 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. MAJOR [ve.ui.WikiaMediaInsertDialog.js#L14](https://github.com/Wikia/app/blob/effd1b6bbd6bbbbf074f5906de5f49e14fa60f42/extensions/VisualEditor/wikia/modules/ve/ui/dialogs/ve.ui.WikiaMediaInsertDialog.js#L14): Make this function anonymous by removing its name: 'function() {...}'. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3ANamedFunctionExpression)
1. MAJOR [ve.ui.WikiaMediaInsertDialog.js#L273](https://github.com/Wikia/app/blob/effd1b6bbd6bbbbf074f5906de5f49e14fa60f42/extensions/VisualEditor/wikia/modules/ve/ui/dialogs/ve.ui.WikiaMediaInsertDialog.js#L273): Remove this commented out code. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3ACommentedCode)
1. MAJOR [ve.ui.WikiaMediaInsertDialog.js#L625](https://github.com/Wikia/app/blob/effd1b6bbd6bbbbf074f5906de5f49e14fa60f42/extensions/VisualEditor/wikia/modules/ve/ui/dialogs/ve.ui.WikiaMediaInsertDialog.js#L625): Restrict what this loop acts on by testing each property. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AForIn)
1. MAJOR [ve.ui.WikiaMediaInsertDialog.js#L650](https://github.com/Wikia/app/blob/effd1b6bbd6bbbbf074f5906de5f49e14fa60f42/extensions/VisualEditor/wikia/modules/ve/ui/dialogs/ve.ui.WikiaMediaInsertDialog.js#L650): Restrict what this loop acts on by testing each property. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AForIn)
1. MAJOR [ve.ui.WikiaMediaInsertDialog.js#L653](https://github.com/Wikia/app/blob/effd1b6bbd6bbbbf074f5906de5f49e14fa60f42/extensions/VisualEditor/wikia/modules/ve/ui/dialogs/ve.ui.WikiaMediaInsertDialog.js#L653): Remove this assignment of "type". [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AS2310)
1. MAJOR [ve.ui.WikiaVideoInsertDialog.js#L14](https://github.com/Wikia/app/blob/effd1b6bbd6bbbbf074f5906de5f49e14fa60f42/extensions/VisualEditor/wikia/modules/ve/ui/dialogs/ve.ui.WikiaVideoInsertDialog.js#L14): Make this function anonymous by removing its name: 'function() {...}'. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3ANamedFunctionExpression)
1. MAJOR [ve.ui.WikiaAnonWarningWidget.js#L1](https://github.com/Wikia/app/blob/effd1b6bbd6bbbbf074f5906de5f49e14fa60f42/extensions/VisualEditor/wikia/modules/ve/ui/widgets/ve.ui.WikiaAnonWarningWidget.js#L1): Make this function anonymous by removing its name: 'function() {...}'. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3ANamedFunctionExpression)
1. MAJOR [ve.ui.WikiaUploadWidget.js#L14](https://github.com/Wikia/app/blob/effd1b6bbd6bbbbf074f5906de5f49e14fa60f42/extensions/VisualEditor/wikia/modules/ve/ui/widgets/ve.ui.WikiaUploadWidget.js#L14): Make this function anonymous by removing its name: 'function() {...}'. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3ANamedFunctionExpression)
1. MINOR [VisualEditor.php#L45](https://github.com/Wikia/app/blob/effd1b6bbd6bbbbf074f5906de5f49e14fa60f42/extensions/VisualEditor/wikia/VisualEditor.php#L45): Define a constant instead of duplicating this literal "scripts" 3 times. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1192)
1. MINOR [VisualEditor.php#L46](https://github.com/Wikia/app/blob/effd1b6bbd6bbbbf074f5906de5f49e14fa60f42/extensions/VisualEditor/wikia/VisualEditor.php#L46): Define a constant instead of duplicating this literal "dependencies" 3 times. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1192)

**Comment by WikiaTech** [comment]
SonarQube analysis reported 11 issues
* MAJOR 8 major
* MINOR 2 minor
* INFO 1 info

#### Top 10 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. MAJOR [ve.ui.WikiaMediaInsertDialog.js#L14](https://github.com/Wikia/app/blob/22ba81189375e2ef3e589824dee7dc97ad3cf814/extensions/VisualEditor/wikia/modules/ve/ui/dialogs/ve.ui.WikiaMediaInsertDialog.js#L14): Make this function anonymous by removing its name: 'function() {...}'. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3ANamedFunctionExpression)
1. MAJOR [ve.ui.WikiaMediaInsertDialog.js#L273](https://github.com/Wikia/app/blob/22ba81189375e2ef3e589824dee7dc97ad3cf814/extensions/VisualEditor/wikia/modules/ve/ui/dialogs/ve.ui.WikiaMediaInsertDialog.js#L273): Remove this commented out code. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3ACommentedCode)
1. MAJOR [ve.ui.WikiaMediaInsertDialog.js#L625](https://github.com/Wikia/app/blob/22ba81189375e2ef3e589824dee7dc97ad3cf814/extensions/VisualEditor/wikia/modules/ve/ui/dialogs/ve.ui.WikiaMediaInsertDialog.js#L625): Restrict what this loop acts on by testing each property. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AForIn)
1. MAJOR [ve.ui.WikiaMediaInsertDialog.js#L650](https://github.com/Wikia/app/blob/22ba81189375e2ef3e589824dee7dc97ad3cf814/extensions/VisualEditor/wikia/modules/ve/ui/dialogs/ve.ui.WikiaMediaInsertDialog.js#L650): Restrict what this loop acts on by testing each property. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AForIn)
1. MAJOR [ve.ui.WikiaMediaInsertDialog.js#L653](https://github.com/Wikia/app/blob/22ba81189375e2ef3e589824dee7dc97ad3cf814/extensions/VisualEditor/wikia/modules/ve/ui/dialogs/ve.ui.WikiaMediaInsertDialog.js#L653): Remove this assignment of "type". [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AS2310)
1. MAJOR [ve.ui.WikiaVideoInsertDialog.js#L14](https://github.com/Wikia/app/blob/22ba81189375e2ef3e589824dee7dc97ad3cf814/extensions/VisualEditor/wikia/modules/ve/ui/dialogs/ve.ui.WikiaVideoInsertDialog.js#L14): Make this function anonymous by removing its name: 'function() {...}'. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3ANamedFunctionExpression)
1. MAJOR [ve.ui.WikiaAnonWarningWidget.js#L1](https://github.com/Wikia/app/blob/22ba81189375e2ef3e589824dee7dc97ad3cf814/extensions/VisualEditor/wikia/modules/ve/ui/widgets/ve.ui.WikiaAnonWarningWidget.js#L1): Make this function anonymous by removing its name: 'function() {...}'. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3ANamedFunctionExpression)
1. MAJOR [ve.ui.WikiaUploadWidget.js#L14](https://github.com/Wikia/app/blob/22ba81189375e2ef3e589824dee7dc97ad3cf814/extensions/VisualEditor/wikia/modules/ve/ui/widgets/ve.ui.WikiaUploadWidget.js#L14): Make this function anonymous by removing its name: 'function() {...}'. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3ANamedFunctionExpression)
1. MINOR [VisualEditor.php#L45](https://github.com/Wikia/app/blob/22ba81189375e2ef3e589824dee7dc97ad3cf814/extensions/VisualEditor/wikia/VisualEditor.php#L45): Define a constant instead of duplicating this literal "scripts" 3 times. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1192)
1. MINOR [VisualEditor.php#L46](https://github.com/Wikia/app/blob/22ba81189375e2ef3e589824dee7dc97ad3cf814/extensions/VisualEditor/wikia/VisualEditor.php#L46): Define a constant instead of duplicating this literal "dependencies" 3 times. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1192)

**Pull request merged**